MAVProxy/modules/mavproxy_wp.py

```diff
@@ -190,9 +191,16 @@ def idle_task(self):
         if self.wp_period.trigger():
             # cope with packet loss fetching mission
             if self.master is not None and self.master.time_since('MISSION_ITEM') >= 2 and self.wploader.count() < getattr(self.wploader,'expected_count',0):
-                wps = self.missing_wps_to_request();
-                print("re-requesting WPs %s" % str(wps))
-                self.send_wp_requests(wps)
+                self.wploader.retry_count = self.wploader.retry_count + 1 
+                if self.wploader.retry_count > 5:
```

Comment:
this will avoid that annoying retry forever, but it will mean that if you are on a lossy link with lots of WPs then you'll never complete as it will keep restarting